pandas/core/base.py

```diff
@@ -1228,7 +1228,10 @@ def _map_values(self, mapper, na_action=None):
                 # possibility that they are tuples
                 from pandas import Series
 
-                mapper = Series(mapper)
+                if not mapper:
```

Comment:
Same question: what user-code hits this?

Reply:
This is raised in `pandas/tests/series/test_apply.py::TestSeriesMap::test_map_empty`. So for instance, the following code would raise this warning:
```
index = Index(list(range(k)), name=name)
s = Series(index)
result = s.map({})
```
where `result` is asserted against `pd.Series(np.nan, index=s.index)`. Since np.nan will dictate the dtype to be float, the empty mapper also has to have the dtype float64.